stable/dex/templates/config-openssl.yaml

```diff
@@ -8,10 +8,6 @@ metadata:
     heritage: "{{ .Release.Service }}"
     release: "{{ .Release.Name }}"
   name: {{ template "dex.fullname" . }}-openssl-config
-  annotations:
```

Comment:
Why these changes to dex? Could you provide some context, just want to know why we are no longer using this hook, its a fairly large change so want to be careful.

Reply:
The context is in the commit message for this commit. Basically this causes the configmap to be be deleted immediatly after creation, and the `grpc-tls` and `web-tls` jobs which depend on it fail, causing the install to fail. This is the backport from https://github.com/helm/charts/pull/18380, but since the charts have diverged enough I couldn't just `cherry-pick` it ;(.